src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesAppInstanceStatus.java

```diff
@@ -45,8 +45,8 @@ public KubernetesAppInstanceStatus(String moduleId, Pod pod, KubernetesDeployerP
 		this.moduleId = moduleId;
 		this.pod = pod;
 		this.properties = properties;
-		// we assume one container per pod
-		if (pod != null && pod.getStatus().getContainerStatuses().size() == 1) {
+		// we assume one container per pod - if more than one the status is for the first one
+		if (pod != null && pod.getStatus().getContainerStatuses().size() >= 1) {
```

Comment:
I guess this is where it gets tricky. Since the entire stream deployment status is computed based on the `ContainerStatus` I think we need to get consider checking all the ContainerStatus info. Otherwise, the stream status would be considered `deployed` even if some of the requested app instances aren't deployed yet.